src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -52,6 +70,341 @@ std::string TPCHQueryGenerator::get_preparation_queries() const {
+
+      parameters.emplace_back("'"s + begin_date + "'");
+      parameters.emplace_back("'"s + end_date + "'");
+      break;
+    }
+
+    case 15 - 1: {
+      auto query_15 = std::string{tpch_queries.at(15)};
+
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 4 * 12 + 9};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff + 3);
+
+      // Hack: We cannot use prepared statements in TPC-H 15. Thus, we need to build the SQL string by hand.
+      // By manually replacing the `?` from tpch_queries.cpp, we can keep all queries in a readable form there.
+      // This is ugly, but at least we can assert that nobody tampered with the string over there.
+      static constexpr auto BEGIN_DATE_OFFSET = 156;
+      static constexpr auto END_DATE_OFFSET = 192;
+      DebugAssert((std::string_view{&query_15[BEGIN_DATE_OFFSET], 10} == "1996-01-01" &&  // NOLINT
+                   std::string_view{&query_15[END_DATE_OFFSET], 10} == "1996-04-01"),     // NOLINT
+                  "TPC-H 15 string has been modified");
+      query_15.replace(BEGIN_DATE_OFFSET, 10, begin_date);
+      query_15.replace(END_DATE_OFFSET, 10, end_date);
+
+      boost::replace_all(query_15, std::string("revenueview"), std::string("revenue") + std::to_string(_q15_view_id++));
+
+      // Not using _build_executable_query here
+      return query_15;
+    }
+
+    case 16 - 1: {
+      const auto brand = brand_char_dist(random_engine) * 10 + brand_char_dist(random_engine);
+
+      const auto full_type = std::string{p_types_set.list[type_dist(random_engine)].text};
+      const auto partial_type = std::string(full_type, 0, full_type.find_last_of(' '));
+
+      auto sizes_copy = sizes;
+      std::shuffle(sizes_copy.begin(), sizes_copy.end(), random_engine);
+
+      parameters.emplace_back("'Brand#"s + std::to_string(brand) + "'");
+      parameters.emplace_back("'"s + partial_type + "%'");
+      for (auto i = 0; i < 8; ++i) parameters.emplace_back(std::to_string(sizes_copy[i]));
+      break;
+    }
+
+    case 17 - 1: {
+      const auto brand = brand_char_dist(random_engine) * 10 + brand_char_dist(random_engine);
+      const auto container = p_cntr_set.list[nation_dist(random_engine)].text;
+
+      parameters.emplace_back("'Brand#"s + std::to_string(brand) + "'");
+      parameters.emplace_back("'"s + container + "'");
+      break;
+    }
+
+    case 18 - 1: {
+      static thread_local std::uniform_int_distribution<> quantity_dist{312, 315};
+      const auto quantity = quantity_dist(random_engine);
+
+      parameters.emplace_back(std::to_string(quantity));
+      break;
+    }
+
+    case 19 - 1: {
+      static thread_local std::uniform_int_distribution<> quantity1_dist{1, 10};
```

Comment:
std::uniform_int_distribution boils down to a pair of ints and will probably be optimized away entirely. Probably not worth making this thread_local and confuse people (me at least. a bit.)

Reply:
The documentation of this is unusually bad. Looks like you don't need internal state to guarantee the distribution: https://stackoverflow.com/questions/30103356/distributions-and-internal-state